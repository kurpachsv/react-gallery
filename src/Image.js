import PropTypes from 'prop-types';
import React from 'react';
import styles from './image.css';

const Image = ({
    style,
    className,
    src,
    alt,
    visible,
    height,
    width,
    newWidth,
    newHeight,
    newWidthInPercent,
    placeholderHeight,
    enableMasonry,
    maxHeight,
    maxWidth,
    fixedBottom,
    ...rest
}) => {
    const ratio = width / height;
    return (
        <img
            {...rest}
            className={`${styles.image} ${className}`}
            src={visible ? src : null}
            alt={alt}
            style={{
                display: visible ? null : 'none',
                position: 'absolute',
                width: ratio > 1 ? '100%' : 'auto',
                height: ratio > 1 ? 'auto' : `calc(100% - ${fixedBottom}px)`,
                ...style,
            }}
        />
    );
};

Image.propTypes = {
    className: PropTypes.string,
    src: PropTypes.string.isRequired,
    alt: PropTypes.string,
    visible: PropTypes.bool,
};

Image.defaultProps = {
    className: '',
    alt: '',
    visible: true,
};

export default Image;
