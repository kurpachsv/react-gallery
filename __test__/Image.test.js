import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import Image from '../src/Image';

describe('Image', () => {
    it('should render correctly', () => {
        const output = shallow(
            <Image
                src="mockUrl"
                alt="mockTitle"
            />
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
