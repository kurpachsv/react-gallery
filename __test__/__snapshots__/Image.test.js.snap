// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render correctly 1`] = `
<img
  alt="mockTitle"
  className="image"
  src="mockUrl"
/>
`;
